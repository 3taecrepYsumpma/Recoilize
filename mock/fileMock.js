// mock used for Jest config in package.json
module.exports = 'test-file-stub';