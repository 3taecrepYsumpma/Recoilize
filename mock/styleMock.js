// mock for jest CSS files, used in package.json for jest config
module.exports = {};