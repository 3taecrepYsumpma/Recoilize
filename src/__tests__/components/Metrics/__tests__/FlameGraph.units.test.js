import React from 'react';
import '@testing-library/jest-dom'
import { 
  cleanup, 
  render, 
  screen 
} from '@testing-library/react'; // custom testing functions necessary to pass tests

import FlameGraph from '../../../../app/components/Metrics/FlameGraph'; // component being tested
import { snapshotHistoryMock } from '../../../../../mock/state-snapshot'; // this is our mock state that we will use to run our tests
import { componentAtomTreeMock } from '../../../../../mock/snapshot.js';


const mock = snapshotHistoryMock.cleanComponentAtomTree;
const mockHeight = 100
const mockWidth = 50


afterEach(cleanup);

describe('Flame graph displays correct information', () => {
  it('should render cleanedComponentAtomTree, width, and height', () => {
    const {asFragment} = render(
      <FlameGraph 
        cleanedComponentAtomTree={mock}
        width={mockWidth}
        height={mockHeight}
      />
    )
    expect(asFragment()).toMatchSnapshot();
  });
  
  //check that labels are being properly passed in 
  it('should contain cleanedComponentAtomTree', () => {
    // render flame graph
    render(
      <FlameGraph 
        cleanedComponentAtomTree={mock}
        width={mockWidth}
        height={mockHeight}
      />
    )
    //ensure that a name property is being passed in and can be found 
    const board = screen.queryByText('Board');
    const row = screen.queryAllByText('Row');
    const box = screen.queryAllByText('Box');
    
    expect(board).toBeVisible();
    expect(box[0]).toBeVisible();
    expect(row[0]).toBeVisible();

    //the mock data should also include an actualDuration with a number value
    expect.objectContaining({actualDuration: expect.any(Number)})
    //ensure recoilNode properties are passed in 
    expect.objectContaining({recoilNode: expect.any(String)})
  });

});

//is passing, but should not be because nothing should render when no data is passed in
describe('Flame graph initial rendering', () => {
  it('should not render when no values are passed in', () => {
    const {asFragment} = render(
      <FlameGraph 
        cleanedComponentAtomTree={componentAtomTreeMock}
        width={mockWidth}
        height={mockHeight}
      />
    )
    expect(asFragment()).toMatchSnapshot();
  });
});