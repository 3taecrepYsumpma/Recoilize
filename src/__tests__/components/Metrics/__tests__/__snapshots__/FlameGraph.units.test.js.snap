// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flame graph displays correct information should render cleanedComponentAtomTree, width, and height 1`] = `
<DocumentFragment>
  <svg
    height="100"
    width="50"
  >
    <g
      class="visx-group"
      transform="translate(0, 0)"
    >
      <g
        transform="translate(1, 1)"
      >
        <rect
          fill="rgb(238, 161, 57)"
          fill-opacity="1"
          height="98"
          id="rect-0"
          width="11"
        />
        <clippath
          id="clip-0"
        >
          <use
            xlink:href="#rect-0"
          />
        </clippath>
        <text
          clip-path="url(#clip-0)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          PlaygroundRender
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             1.80
          </tspan>
        </text>
      </g>
      <g
        transform="translate(13, 1)"
      >
        <rect
          fill="rgb(234, 193, 224)"
          fill-opacity="1"
          height="55.99999999999999"
          id="rect-1"
          width="11"
        />
        <clippath
          id="clip-1"
        >
          <use
            xlink:href="#rect-1"
          />
        </clippath>
        <text
          clip-path="url(#clip-1)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          Board
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             1.06
          </tspan>
        </text>
      </g>
      <g
        transform="translate(25, 1)"
      >
        <rect
          fill="rgb(248, 237, 255)"
          fill-opacity="1"
          height="16"
          id="rect-2"
          width="12"
        />
        <clippath
          id="clip-2"
        >
          <use
            xlink:href="#rect-2"
          />
        </clippath>
        <text
          clip-path="url(#clip-2)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          Row
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             0.40
          </tspan>
        </text>
      </g>
      <g
        transform="translate(25, 18)"
      >
        <rect
          fill="rgb(253, 251, 255)"
          fill-opacity="1"
          height="8"
          id="rect-3"
          width="12"
        />
        <clippath
          id="clip-3"
        >
          <use
            xlink:href="#rect-3"
          />
        </clippath>
        <text
          clip-path="url(#clip-3)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          Row
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             0.21
          </tspan>
        </text>
      </g>
      <g
        transform="translate(25, 27)"
      >
        <rect
          fill="rgb(254, 253, 255)"
          fill-opacity="1"
          height="6"
          id="rect-4"
          width="12"
        />
        <clippath
          id="clip-4"
        >
          <use
            xlink:href="#rect-4"
          />
        </clippath>
        <text
          clip-path="url(#clip-4)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          Row
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             0.17
          </tspan>
        </text>
      </g>
      <g
        transform="translate(38, 1)"
      >
        <rect
          fill="rgb(254, 253, 255)"
          fill-opacity="1"
          height="3"
          id="rect-5"
          width="11"
        />
        <clippath
          id="clip-5"
        >
          <use
            xlink:href="#rect-5"
          />
        </clippath>
        <text
          clip-path="url(#clip-5)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          Box
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             0.18
          </tspan>
        </text>
      </g>
      <g
        transform="translate(38, 5)"
      >
        <rect
          fill="rgb(255, 255, 255)"
          fill-opacity="1"
          height="1"
          id="rect-6"
          width="11"
        />
        <clippath
          id="clip-6"
        >
          <use
            xlink:href="#rect-6"
          />
        </clippath>
        <text
          clip-path="url(#clip-6)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          Box
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             0.08
          </tspan>
        </text>
      </g>
      <g
        transform="translate(38, 7.000000000000001)"
      >
        <rect
          fill="rgb(255, 255, 255)"
          fill-opacity="1"
          height="0"
          id="rect-7"
          width="11"
        />
        <clippath
          id="clip-7"
        >
          <use
            xlink:href="#rect-7"
          />
        </clippath>
        <text
          clip-path="url(#clip-7)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          Box
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             0.07
          </tspan>
        </text>
      </g>
      <g
        transform="translate(38, 18)"
      >
        <rect
          fill="rgb(255, 255, 255)"
          fill-opacity="1"
          height="0"
          id="rect-8"
          width="11"
        />
        <clippath
          id="clip-8"
        >
          <use
            xlink:href="#rect-8"
          />
        </clippath>
        <text
          clip-path="url(#clip-8)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          Box
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             0.06
          </tspan>
        </text>
      </g>
      <g
        transform="translate(38, 19)"
      >
        <rect
          fill="rgb(255, 255, 255)"
          fill-opacity="1"
          height="0"
          id="rect-9"
          width="11"
        />
        <clippath
          id="clip-9"
        >
          <use
            xlink:href="#rect-9"
          />
        </clippath>
        <text
          clip-path="url(#clip-9)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          Box
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             0.06
          </tspan>
        </text>
      </g>
      <g
        transform="translate(38, 20)"
      >
        <rect
          fill="rgb(255, 255, 255)"
          fill-opacity="1"
          height="1"
          id="rect-10"
          width="11"
        />
        <clippath
          id="clip-10"
        >
          <use
            xlink:href="#rect-10"
          />
        </clippath>
        <text
          clip-path="url(#clip-10)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          Box
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             0.06
          </tspan>
        </text>
      </g>
      <g
        transform="translate(38, 27)"
      >
        <rect
          fill="rgb(255, 255, 255)"
          fill-opacity="1"
          height="0"
          id="rect-11"
          width="11"
        />
        <clippath
          id="clip-11"
        >
          <use
            xlink:href="#rect-11"
          />
        </clippath>
        <text
          clip-path="url(#clip-11)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          Box
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             0.06
          </tspan>
        </text>
      </g>
      <g
        transform="translate(38, 28.000000000000004)"
      >
        <rect
          fill="rgb(255, 255, 255)"
          fill-opacity="1"
          height="0"
          id="rect-12"
          width="11"
        />
        <clippath
          id="clip-12"
        >
          <use
            xlink:href="#rect-12"
          />
        </clippath>
        <text
          clip-path="url(#clip-12)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          Box
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             0.05
          </tspan>
        </text>
      </g>
      <g
        transform="translate(38, 28.999999999999996)"
      >
        <rect
          fill="rgb(255, 255, 255)"
          fill-opacity="1"
          height="0"
          id="rect-13"
          width="11"
        />
        <clippath
          id="clip-13"
        >
          <use
            xlink:href="#rect-13"
          />
        </clippath>
        <text
          clip-path="url(#clip-13)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          Box
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             0.04
          </tspan>
        </text>
      </g>
    </g>
  </svg>
</DocumentFragment>
`;

exports[`Flame graph initial rendering should not render when no values are passed in 1`] = `
<DocumentFragment>
  <svg
    height="100"
    width="50"
  >
    <g
      class="visx-group"
      transform="translate(0, 0)"
    >
      <g
        transform="translate(1, 1)"
      >
        <rect
          fill-opacity="1"
          height="98"
          id="rect-0"
          width="8"
        />
        <clippath
          id="clip-0"
        >
          <use
            xlink:href="#rect-0"
          />
        </clippath>
        <text
          clip-path="url(#clip-0)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(10, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-1"
          width="9"
        />
        <clippath
          id="clip-1"
        >
          <use
            xlink:href="#rect-1"
          />
        </clippath>
        <text
          clip-path="url(#clip-1)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(10, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-2"
          width="9"
        />
        <clippath
          id="clip-2"
        >
          <use
            xlink:href="#rect-2"
          />
        </clippath>
        <text
          clip-path="url(#clip-2)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(10, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-3"
          width="9"
        />
        <clippath
          id="clip-3"
        >
          <use
            xlink:href="#rect-3"
          />
        </clippath>
        <text
          clip-path="url(#clip-3)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(20, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-4"
          width="8.999999999999996"
        />
        <clippath
          id="clip-4"
        >
          <use
            xlink:href="#rect-4"
          />
        </clippath>
        <text
          clip-path="url(#clip-4)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(20, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-5"
          width="8.999999999999996"
        />
        <clippath
          id="clip-5"
        >
          <use
            xlink:href="#rect-5"
          />
        </clippath>
        <text
          clip-path="url(#clip-5)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(20, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-6"
          width="8.999999999999996"
        />
        <clippath
          id="clip-6"
        >
          <use
            xlink:href="#rect-6"
          />
        </clippath>
        <text
          clip-path="url(#clip-6)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(20, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-7"
          width="8.999999999999996"
        />
        <clippath
          id="clip-7"
        >
          <use
            xlink:href="#rect-7"
          />
        </clippath>
        <text
          clip-path="url(#clip-7)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(30, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-8"
          width="9"
        />
        <clippath
          id="clip-8"
        >
          <use
            xlink:href="#rect-8"
          />
        </clippath>
        <text
          clip-path="url(#clip-8)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(30, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-9"
          width="9"
        />
        <clippath
          id="clip-9"
        >
          <use
            xlink:href="#rect-9"
          />
        </clippath>
        <text
          clip-path="url(#clip-9)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(40, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-10"
          width="9"
        />
        <clippath
          id="clip-10"
        >
          <use
            xlink:href="#rect-10"
          />
        </clippath>
        <text
          clip-path="url(#clip-10)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
    </g>
  </svg>
</DocumentFragment>
`;
